source/globalCommands.py

```diff
@@ -2410,3 +2414,113 @@ def script_recognizeWithUwpOcr(self, gesture):
 #: The single global commands instance.
 #: @type: L{GlobalCommands}
 commands = GlobalCommands()
+
+class ConfigProfileActivationCommands(ScriptableObject):
+	"""Singleton scriptable object that collects scripts for available configuration profiles."""
+
+	scriptCategory = SCRCAT_CONFIG_PROFILES
+
+	@classmethod
+	def __new__(cls, *args, **kwargs):
+		# Iterate through the available profiles, creating scripts for them.
+		for profile in config.conf.listProfiles():
+			cls.addScriptForProfile(profile)
+		return 		super(ConfigProfileActivationCommands, cls).__new__(cls)
+
+	@classmethod
+	def _getScriptNameForProfile(cls, name):
+		invalidChars = set()
+		for c in name:
+			if not c.isalnum() and c != "_":
+				invalidChars.add(c)
+		for c in invalidChars:
+			name=name.replace(c, b16encode(c))

```

Comment:
I don't think that b16encode works for non-ascii values (I.e. any other unicode value).
Someone sent NV Access an email showing the following traceback from a very recent alpha snapshot:
```
ERROR - keyboardHandler.internal_keyDownEvent (14:59:07.003): 
internal_keyDownEvent 
Traceback (most recent call last): 
  File "keyboardHandler.pyc", line 178, in internal_keyDownEvent 
  File "inputCore.pyc", line 421, in executeGesture 
  File "baseObject.pyc", line 47, in __get__ 
  File "baseObject.pyc", line 147, in _getPropertyViaCache 
  File "inputCore.pyc", line 145, in _get_script 
  File "scriptHandler.pyc", line 68, in findScript 
  File "globalCommands.pyc", line 2525, in <module> 
  File "globalCommands.pyc", line 2427, in __new__ 
  File "globalCommands.pyc", line 2466, in addScriptForProfile 
  File "globalCommands.pyc", line 2438, in _getScriptNameForProfile 
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf1' in position 12: ordinal not in range(128) 
```